Source/WebCore/platform/network/cocoa/RangeResponseGenerator.mm

```diff
@@ -96,7 +96,11 @@ static ResourceResponse synthesizedResponseForRange(const ResourceResponse& orig
 void RangeResponseGenerator::removeTask(WebCoreNSURLSessionDataTask *task)
 {
     ASSERT(isMainThread());
-    auto* data = m_map.get(task.originalRequest.URL.absoluteString);
+    auto urlString = task.originalRequest.URL.absoluteString;
+    // HashMap::get() crashes if a null String is passed.
+    if (!urlString)
```

Comment:
This doesn't look great. We may end up leaving tasks in m_map that we don't want to.

Why do we end up with a null String here? How come were were able to add the task to m_map in the first place (if its urlString was originally null)?

Reply:
CC @achristensen07

Reply:
`RangeResponseGenerator::willSynthesizeRangeResponses()` calls:
```
 m_map.ensure(originalRequest.URL.absoluteString, [&] {
```

with `originalRequest` being `task.originalRequest`.

So if `task.originalRequest.URL.absoluteString` was nil, we would have crashed on adding the task to the map it seems. Something else must be going on here.

Reply:
Oh, `RangeResponseGenerator::willSynthesizeRangeResponses()` does have this check:
```
    if (!originalRequest.URL)
        return false;
```

So the issue may be that we create the task (with a null URL for some reason), we call willSynthesizeRangeResponses() which will return early and not add the task to m_map. Then we cancel the task, which calls `removeTask()` but we don't have the same null URL check there.

Given that, I think the patch is OK. The task wouldn't be in m_map anyway.